app/test/unit/git/checkout-test.ts

```diff
@@ -40,4 +71,70 @@ describe('git/checkout', () => {
     const validBranch = tip as IValidBranch
     expect(validBranch.branch.name).to.equal('commit-with-long-description')
   })
+
+  it('can checkout a branch when it exists on multiple remotes', async () => {
+    const path = await setupFixtureRepository('checkout-test-cases')
+    const repository = new Repository(path, -1, null, false)
+
+    const expectedBranch = 'first'
+    const firstRemote = 'first-remote'
+    const secondRemote = 'second-remote'
+
+    const branches = await getBranches(repository)
+    const firstBranch = `${firstRemote}/${expectedBranch}`
+    const firstRemoteBranch = branches.find(b => b.name === firstBranch)
+
+    if (firstRemoteBranch == null) {
+      throw new Error(`Could not find branch: '${firstBranch}'`)
+    }
+
+    const secondBranch = `${secondRemote}/${expectedBranch}`
+    const secondRemoteBranch = branches.find(b => b.name === secondBranch)
+
+    if (secondRemoteBranch == null) {
+      throw new Error(`Could not find branch: '${secondBranch}'`)
+    }
+
+    await checkoutBranch(repository, null, firstRemoteBranch)
+
+    const store = new GitStore(repository, shell)
+    await store.loadStatus()
+    const tip = store.tip
+
+    expect(tip.kind).to.equal(TipState.Valid)
+
+    const validBranch = tip as IValidBranch
+    expect(validBranch.branch.name).to.equal(expectedBranch)
+    expect(validBranch.branch.type).to.equal(BranchType.Local)
+    expect(validBranch.branch.remote).to.equal('first-remote')
+  })
+
+  it('will fail when an existing branch matches the remote branch', async () => {
+    const path = await setupFixtureRepository('checkout-test-cases')
+    const repository = new Repository(path, -1, null, false)
+
+    const expectedBranch = 'first'
+    const firstRemote = 'first-remote'
+
+    const branches = await getBranches(repository)
+    const firstBranch = `${firstRemote}/${expectedBranch}`
+    const remoteBranch = branches.find(b => b.name === firstBranch)
+
+    if (remoteBranch == null) {
+      throw new Error(`Could not find branch: '${firstBranch}'`)
+    }
+
+    await createBranch(repository, expectedBranch)
+
+    let errorRaised = false
+
+    try {
+      await checkoutBranch(repository, null, remoteBranch)
+    } catch (error) {
+      errorRaised = true
+      expect(error.message).to.equal('A branch with that name already exists.')
```

Comment:
This is still a known limitation with this approach, and without using namespaces to keep these unique like we might be doing in #3602. But I'm fine with kicking that can further down the road and focusing on the original issue reported.